### DISCUSSION TITLE
Support history in v repl

### DISCUSSION THREAD

**Discussion by bombsimon** [vlang/v#11651] [Ideas]
I often feel the need of stepping in the history while trying something out in the repl. It would be great if the repl supported the up- and down arrows to step backwards and forward in the history like python's repl.

Right now it seems like `[[A` and `[[B` isn't handled at all. I guess this would need to be implemented by extending `readline` to support syscalls (or create a separate keyboard package for events).

**Comment by spytheman** [comment]
For now you can try: `rlwrap --always-readline v repl` .

**Comment by bombsimon** [comment]
Thanks! Yeah, running a readline wrapper is definitely a workaround for now. I'll leave the issue open until this is implemented in `v` natively or if decided not to implement.

**Comment by RiordanIX** [comment]
This probably needs a properly working `termios` module under `term`. This will allow putting the terminal into raw mode so repl can handle all keyboard inputs.

**Comment by serkonda7** [comment]
This was recently fixed and should work fine after updating V.

**Comment by bombsimon** [comment]
> This was recently fixed and should work fine after updating V.

Do I need to do anything special? Arrow up/down still just sends the sequence for me:

```v
❯ v
Welcome to the V REPL (for help with V itself, type `exit`, then run `v help`).
V 0.2.1 c66ee8a
Use Ctrl-C or `exit` to exit, or `help` to see other available commands
>>> println("hello")
hello
>>> ^[[A^[[B
```

macOS 11.1
iTerm2 3.4.3

**Comment by serkonda7** [comment]
You should run `v up` adn try again

**Comment by adelarsq** [comment]
@serkonda7 doesn't work for me after `v up` with iTerm 3.4.3 and macOS 10.15.7

**Comment by serkonda7** [comment]
@adelarsq could you please paste the output of `v doctor`.

**Comment by adelarsq** [comment]
@serkonda7 yes. For `v doctor` shows:

```
adelar@MacBook-Pro ~ % v doctor
OS: macos, Mac OS X, 10.15.7, 19H114
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i7-3540M CPU @ 3.00GHz
CC version: Apple clang version 12.0.0 (clang-1200.0.32.28)

getwd: /Users/adelar
vmodules: /Users/adelar/.vmodules
vroot: /Users/adelar/dev/v
vexe: /Users/adelar/dev/v/v
vexe mtime: 2021-01-05 14:24:20
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.1 e192773.eaba21d

Git version: git version 2.24.3 (Apple Git-128)
Git vroot status: 0.2-299-geaba21d8
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02
```

For `v repl` pressing down and up shows:

```
adelar@MacBook-Pro ~ % v repl
V 0.2.1 eaba21d
Use Ctrl-C or `exit` to exit, or `help` to see other available commands
>>> a := 2
>>> ^[[A^[[A^[[B^[[B
```

**Comment by serkonda7** [comment]
Oh, I looked into https://github.com/vlang/v/tree/master/vlib/readline and indeed it lacks a macos implementation.
It works only on linux.

**Comment by Delta456** [comment]
Sounds great

**Comment by kjarex** [comment]
As history is obviously meant to be supported already (since it is supported for other platforms and mentioned as TODO in source) it should be labelled as a bug imo
